From a8864e7c8d85b29b39c63ddeed7e7630d8776e50 Mon Sep 17 00:00:00 2001
From: Felix Abecassis <felix.abecassis@gmail.com>
Date: Tue, 4 Mar 2014 19:11:44 +0100
Subject: [PATCH] mediacodec: move timeout from dequeueInputBuffer to
 dequeueOutputBuffer

dequeueOutputBuffer should have an higher priority since releasing an
output buffer often allows MediaCodec to make progress

Conflicts:
	modules/codec/omxil/android_mediacodec.c
---
 modules/codec/omxil/android_mediacodec.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/modules/codec/omxil/android_mediacodec.c b/modules/codec/omxil/android_mediacodec.c
index 47311a6..4292ae1 100644
--- a/modules/codec/omxil/android_mediacodec.c
+++ b/modules/codec/omxil/android_mediacodec.c
@@ -756,12 +756,12 @@ static void InvalidateAllPictures(decoder_t *p_dec)
     vlc_mutex_unlock(get_android_opaque_mutex());
 }
 
-static void GetOutput(decoder_t *p_dec, JNIEnv *env, picture_t **pp_pic)
+static void GetOutput(decoder_t *p_dec, JNIEnv *env, picture_t **pp_pic, jlong timeout)
 {
     decoder_sys_t *p_sys = p_dec->p_sys;
     while (1) {
         int index = (*env)->CallIntMethod(env, p_sys->codec, p_sys->dequeue_output_buffer,
-                                          p_sys->buffer_info, (jlong) 0);
+                                          p_sys->buffer_info, timeout);
         if ((*env)->ExceptionOccurred(env)) {
             (*env)->ExceptionClear(env);
             p_sys->error_state = true;
@@ -978,14 +978,14 @@ static picture_t *DecodeVideo(decoder_t *p_dec, block_t **pp_block)
     const int max_polling_attempts = 50;
     int attempts = 0;
     while (true) {
-        int index = (*env)->CallIntMethod(env, p_sys->codec, p_sys->dequeue_input_buffer, timeout);
+        int index = (*env)->CallIntMethod(env, p_sys->codec, p_sys->dequeue_input_buffer, (jlong) 0);
         if ((*env)->ExceptionOccurred(env)) {
             (*env)->ExceptionClear(env);
             p_sys->error_state = true;
             break;
         }
         if (index < 0) {
-            GetOutput(p_dec, env, &p_pic);
+            GetOutput(p_dec, env, &p_pic, timeout);
             if (p_pic) {
                 /* If we couldn't get an available input buffer but a
                  * decoded frame is available, we return the frame
@@ -1039,7 +1039,7 @@ static picture_t *DecodeVideo(decoder_t *p_dec, block_t **pp_block)
         break;
     }
     if (!p_pic)
-        GetOutput(p_dec, env, &p_pic);
+        GetOutput(p_dec, env, &p_pic, 0);
 
     block_Release(p_block);
     *pp_block = NULL;
-- 
1.8.5.3

